cmd/juju/storage/import.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package storage
+
+import (
+	"regexp"
+
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/jujuclient"
+	"github.com/juju/juju/storage"
+)
+
+// NewImportFilesystemCommand returns a command used to import a filesystem.
+//
+// newStorageImporter is the function to use to acquire a StorageImporter.
+// A non-nil function must be provided.
+//
+// store is an optional ClientStore to use for interacting with the client
+// model/controller storage. If nil, the default file-based store will be
+// used.
+func NewImportFilesystemCommand(
+	newStorageImporter NewStorageImporterFunc,
+	store jujuclient.ClientStore,
+) cmd.Command {
+	cmd := &importFilesystemCommand{}
+	cmd.newAPIFunc = newStorageImporter
+	if store != nil {
+		cmd.SetClientStore(store)
+	}
+	return modelcmd.Wrap(cmd)
+}
+
+// NewStorageImporterFunc is the type of a function passed to
+// NewImportFilesystemCommand, in order to acquire a StorageImporter.
+type NewStorageImporterFunc func(*StorageCommandBase) (StorageImporter, error)
+
+// NewStorageImporter returns a new StorageImporter,
+// given a StorageCommandBase.
+func NewStorageImporter(cmd *StorageCommandBase) (StorageImporter, error) {
+	return cmd.NewStorageAPI()
+}
+
+const (
+	importFilesystemCommandDoc = `
+Import an existing filesystem into the model. This will lead to the model
+taking ownership of the storage, so you must take care not to import storage
+that is in use by another Juju model.
+
+To import a filesystem, you must specify three things:
+
+ - the storage pool in which to look for the storage
```

Comment:
Sorta? It's also used to pick the storage provider.

I'm not really happy with our "pool" terminology, because the pool is pretty much irrelevant after the storage is created. It's just configuration for how to create storage.

Reworded, and included the bit about association.